File: codeforce/A82.cpp
// coder         : Himanshu Mishra (himanshu.m786@gmail.com)
// about program : A82

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <numeric>
#include <cstdlib>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::vector;
using namespace std;

int main() {

    int x, temp;
    cin >> x;


    //string a = "Sheldon", b = "Leonard", c = "Penny", d = "Rajesh", e = "Howard";

    vector<int> v;
    //v = {"a", "b", "c", "d", "e"};
    v.push_back(1);
    v.push_back(2);
    v.push_back(3);
    v.push_back(4);
    v.push_back(5);

    for (int i = 0; i < x; ++i)
    {
        temp = v[i];
        v.push_back(temp);v.push_back(temp);
    }

    if (temp == 1) {
        cout << "Sheldon" << endl;
    }
    else {
        if (temp == 2) {
        cout << "Leonard" << endl;
        }
        else {
            if (temp == 3) {
                cout << "Penny" << endl;
            }
            else {
                if (temp == 4) {
                    cout << "Rajesh" << endl;
                }
                else {
                    cout << "Howard" << endl;
                }
            }
        }
    }
}

File: codeforce/A82.py
"""
Domain Name : Codeforces.com
Link        : http://codeforces.com/problemset/problem/82/A
Problem Name: A. Double Cola
Method      : implementation, math
Coder       : Himanshu-Mishr(himanshu.m786@gmail.com)
"""
import time

def main():
    '''
    '''
    start = time.time()
    cans = input()
    a = 'Sheldon'
    b = 'Leonard'
    c = 'Penny'
    d = 'Rajesh'
    e = 'Howard'

    queue = ['a', 'b', 'c', 'd', 'e']
    for i in range(int(cans)):
        x = queue[0]
        queue = queue[1:]
        queue.append(x)
        queue.append(x)
    else:print(x)
    print("Time taken :", time.time() - start)
if __name__ == '__main__':
    main()
